pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/relocation/RealtimeSegmentRelocator.java

```diff
@@ -58,51 +58,59 @@ public RealtimeSegmentRelocator(PinotHelixResourceManager pinotHelixResourceMana
   }
 
   @Override
-  public void process(List<String> tables) {
-    runRelocation(tables);
+  protected void preprocess() {
+
+  }
+
+  @Override
+  protected void process(String tableNameWithType) {
+    runRelocation(tableNameWithType);
+  }
+
+  @Override
+  protected void postprocess() {
+
```

Comment:
Not related to this, but should we add some metrics on number of segments relocated?

Reply:
Yes, that is a good idea. Will do so in a future PR